### DISCUSSION TITLE
Cumbersome relations setup

### DISCUSSION THREAD

**Discussion by yaeszlo** [pocketbase/pocketbase#3525] [Q&A]
Hello. I am creating online shop PoC using Pocketbase. 

I have stumbled upon an interesting issue that makes the usage a little unpleasant. 

I've created `products` collection. A product has type. Initially I've used Select field for this purpose, but I discovered quickly that **this prevents me from getting all possible types thru View Collection query.**

So I created a new `product_type` collection instead. And here is what I think could be improved:
1. when I created just one field `type_name` and left `id` empty, I couldn't see what I'm picking as `id` is shown as primary field here
![image](https://github.com/pocketbase/pocketbase/assets/82897497/d617c72e-c435-4a74-b774-5c0c7a7673e3)
If I tried to search by readable name, nothing comes up:
![image](https://github.com/pocketbase/pocketbase/assets/82897497/a67637fe-b65e-4219-a284-119ce96ca8fd)

2. so then I tried to name `id`s descriptively, but minimum ID length is 15 chars so I ended up with this abhorrent looking thing:
![image](https://github.com/pocketbase/pocketbase/assets/82897497/51b6952e-877f-4d33-9d68-ad58df09d38b)

I think a simple thing like specifying product type shouldn't be this much a hassle. In my app I will have to somehow be wary of these forced paddings on types IDs.
Unless there's a better way of achieving what I need? I will be grateful for any help!

Cheers

**Comment by ganigeorgiev** [comment]
I'm having difficulties understanding what actually is the question and what you are trying to do.

>  but I discovered quickly that this prevents me from getting all possible types thru View Collection query

Why? What is the problem that you've stumbled?


> if I tried to search by readable name, nothing comes up

I'm not sure that I understand what do you mean by this. The searchbar allow a "search term" (wildcard LIKE search in all fields) or the same syntax as the API rules and filters, aka. `yourField = "value"`.
_You can test it on the live demo and see that in the "messages" collection in the "author" relation picker you can search for "Doe" and it will return all records that contains "Doe" in any field value._

Again, it is not clear to me from your description what is the desired behavior.

If you want to show the type name in the relation pickers instead of the id, you can mark the type name field in your "product_type" collection as "Presentable":

![presentable_screenshot](https://i.imgur.com/O5CEYoW.png)

**Reply by ganigeorgiev** [reply]
This of course will depend on your actual use case, but if you are asking for suggestions how to structure your collections, one possible configuration if you want the types to be dynamic could be to create:

1. "types" collection with at least a "name" or similar `text` field (_you can mark the fields you want to be shown in the relation listings with the "Presentable" toggle; if none is selected we will search for common field names like "title", "name", "label", etc. or fallback to the "id"_).

2. "products" collection with at least a single (or multiple if a product can have more than 1 type) "type" `relation` field pointing to the "types" collection.

Listing all types can be done with `pb.collection("types").getFullList()`.

Creating a new "products" record can be done with something like:
```js
const types = await pb.collection("types").getFullList();

// create a product associated to the first type
const product = await pb.collection("products").create({
    type: types[0].id,
    // ... set other products fields
})
```

If the above is not what you are looking for, please export your collections configuration as JSON (_Admin UI > Settings > Export collections_) and elaborate on the desired behavior.

**Reply by laszlo1337** [reply]
`Presentable` Option is what solved it for me - relation picker is readable now. Thanks!

> Why? What is the problem that you've stumbled?

Answering this - AFAIK one cannot query DB table schema in SQLite, which is what would be needed if I wanted to get all available types from Select field of `product`. That's why I created a new collection for product types in the first place.

**Reply by ganigeorgiev** [reply]
>  AFAIK one cannot query DB table schema in SQLite, which is what would be needed if I wanted to get all available types from Select field of product

The `select` field values are stored as json in the fields option of `_collections.schema`.

You can extract them using the [SQLite json functions](https://www.sqlite.org/json1.html), but if you plan to store a lot of types then using a dedicated "types" collection would be better as you'll be able to perform more optimized filtering, pagination and add/remove types programmatically.
